isobus/src/can_extended_transport_protocol.cpp

```diff
@@ -295,7 +295,7 @@ namespace isobus
 					    (StateMachineState::RxDataSession == tempSession->state) &&
 					    (messageData[SEQUENCE_NUMBER_DATA_INDEX] == (tempSession->lastPacketNumber + 1)))
 					{
-						for (std::uint8_t i = SEQUENCE_NUMBER_DATA_INDEX; (i < PROTOCOL_BYTES_PER_FRAME) && ((PROTOCOL_BYTES_PER_FRAME * tempSession->lastPacketNumber) + i < tempSession->get_message_data_length()); i++)
+						for (std::uint8_t i = SEQUENCE_NUMBER_DATA_INDEX; (i < PROTOCOL_BYTES_PER_FRAME) && ((tempSession->processedPacketsThisSession * PROTOCOL_BYTES_PER_FRAME) + ((PROTOCOL_BYTES_PER_FRAME * tempSession->lastPacketNumber) + i) < tempSession->get_message_data_length()); i++)
```

Comment:
Something's still not quite right... needs additional testing. Will keep looking at it tomorrow